src/pymatgen/core/periodic_table.py

```diff
@@ -1646,6 +1646,7 @@ def get_el_sp(obj: int | SpeciesLike) -> Element | Species | DummySpecies:
     # If obj is already an Element or Species, return as is
     if isinstance(obj, Element | Species | DummySpecies):
         if getattr(obj, "_is_named_isotope", False):
+            warnings.warn(f"Named isotope {obj.name} would be converted to {obj.symbol}", stacklevel=2)
```

Comment:
I'm not 100% sure if there's any case where a name isotope doesn't have `name`?

https://github.com/materialsproject/pymatgen/blob/31f1e1fb8cc1bb517d59ea8e484965bb641c42a0/src/pymatgen/core/periodic_table.py#L1647-L1649

https://github.com/materialsproject/pymatgen/blob/31f1e1fb8cc1bb517d59ea8e484965bb641c42a0/src/pymatgen/core/periodic_table.py#L1105-L1106